packages/x-date-pickers/src/MonthPicker/MonthPicker.tsx

```diff
@@ -143,76 +148,65 @@ export const MonthPicker = React.forwardRef(function MonthPicker<TDate>(
 
     return utils.getMonth(now);
   }, [now, date, utils, disableHighlightToday]);
-  const [focusedMonth, setFocusedMonth] = React.useState<number>(
-    () => selectedMonth || utils.getMonth(now),
-  );
+  const [focusedMonth, setFocusedMonth] = React.useState(() => selectedMonth || todayMonth);
 
-  const isMonthDisabled = React.useCallback(
-    (month: TDate) => {
-      const firstEnabledMonth = utils.startOfMonth(
-        disablePast && utils.isAfter(now, minDate) ? now : minDate,
-      );
+  const [internalHasFocus, setInternalHasFocus] = useControlled<boolean>({
+    name: 'MonthPicker',
+    state: 'hasFocus',
+    controlled: hasFocus,
+    default: autoFocus,
+  });
 
-      const lastEnabledMonth = utils.startOfMonth(
-        disableFuture && utils.isBefore(now, maxDate) ? now : maxDate,
-      );
+  const changeHasFocus = useEventCallback((newHasFocus: boolean) => {
+    setInternalHasFocus(newHasFocus);
 
-      if (utils.isBefore(month, firstEnabledMonth)) {
-        return true;
-      }
+    if (onFocusedViewChange) {
+      onFocusedViewChange(newHasFocus);
+    }
+  });
 
-      if (utils.isAfter(month, lastEnabledMonth)) {
-        return true;
-      }
+  const isMonthDisabled = useEventCallback((month: TDate) => {
+    const firstEnabledMonth = utils.startOfMonth(
+      disablePast && utils.isAfter(now, minDate) ? now : minDate,
+    );
 
-      if (!shouldDisableMonth) {
-        return false;
-      }
+    const lastEnabledMonth = utils.startOfMonth(
+      disableFuture && utils.isBefore(now, maxDate) ? now : maxDate,
+    );
 
-      return shouldDisableMonth(month);
-    },
-    [disableFuture, disablePast, maxDate, minDate, now, shouldDisableMonth, utils],
-  );
+    if (utils.isBefore(month, firstEnabledMonth)) {
+      return true;
+    }
 
-  const onMonthSelect = (month: number) => {
-    if (readOnly) {
-      return;
+    if (utils.isAfter(month, lastEnabledMonth)) {
+      return true;
     }
 
-    const newDate = utils.setMonth(selectedDateOrToday, month);
-    onChange(newDate, 'finish');
-  };
+    if (!shouldDisableMonth) {
+      return false;
+    }
 
-  const [internalHasFocus, setInternalHasFocus] = useControlled<boolean>({
-    name: 'MonthPicker',
-    state: 'hasFocus',
-    controlled: hasFocus,
-    default: autoFocus,
+    return shouldDisableMonth(month);
   });
 
-  const changeHasFocus = React.useCallback(
-    (newHasFocus: boolean) => {
-      setInternalHasFocus(newHasFocus);
+  const handleMonthSelection = useEventCallback((event: React.MouseEvent, year: number) => {
```

Comment:
Shouldn't it be `month` as previously? 🤔 
```suggestion
  const handleMonthSelection = useEventCallback((event: React.MouseEvent, month: number) => {
```

Reply:
:see_no_evil: